src/Workspaces/Remote/ServiceHub/Services/EncapsulateField/RemoteEncapsulateFieldService.cs

```diff
@@ -8,31 +8,61 @@
 using System.Collections.Immutable;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeCleanup;
 using Microsoft.CodeAnalysis.EncapsulateField;
+using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Remote
 {
     internal sealed class RemoteEncapsulateFieldService : BrokeredServiceBase, IRemoteEncapsulateFieldService
     {
-        internal sealed class Factory : FactoryBase<IRemoteEncapsulateFieldService>
+        internal sealed class Factory : FactoryBase<IRemoteEncapsulateFieldService, IRemoteEncapsulateFieldService.ICallback>
         {
-            protected override IRemoteEncapsulateFieldService CreateService(in ServiceConstructionArguments arguments)
-                => new RemoteEncapsulateFieldService(arguments);
+            protected override IRemoteEncapsulateFieldService CreateService(in ServiceConstructionArguments arguments, RemoteCallback<IRemoteEncapsulateFieldService.ICallback> callback)
+                => new RemoteEncapsulateFieldService(arguments, callback);
         }
 
-        public RemoteEncapsulateFieldService(in ServiceConstructionArguments arguments)
+        private readonly RemoteCallback<IRemoteEncapsulateFieldService.ICallback> _callback;
+
+        public RemoteEncapsulateFieldService(in ServiceConstructionArguments arguments, RemoteCallback<IRemoteEncapsulateFieldService.ICallback> callback)
             : base(arguments)
         {
+            _callback = callback;
+        }
+
+#if TODO // Replace the below specialization with a call to a generic impl once https://github.com/microsoft/vs-streamjsonrpc/issues/789 is fixed
+        private CodeCleanupOptionsProvider GetClientOptionsProvider(RemoteServiceCallbackId callbackId)
```

Comment:
Creates options provider that fetches options from the client and caches them for a given language.